includes/parser/tests/ParserOutputTest.php

```diff
@@ -0,0 +1,382 @@
+<?php
+
+class ParserOutputTest extends WikiaBaseTest {
+
+	/**
+	 * @dataProvider getParserOutputVars
+	 */
+	public function testMergeExternalParserOutputVars( $sourceVars, $externalVars, $expectedVars ) {
+		$parserOutputSource = new ParserOutput();
+		$parserOutputExternal = new ParserOutput();
+
+		foreach ( $parserOutputSource::$varsToMerge as $var ) {
+			$parserOutputSource->$var = $sourceVars[$var];
+			$parserOutputExternal->$var = $externalVars[$var];
+		}
+
+		$parserOutputSource->mergeExternalParserOutputVars( $parserOutputExternal );
+
+		foreach ( $parserOutputSource::$varsToMerge as $var ) {
+			$this->assertSame( $expectedVars[$var], $parserOutputSource->$var );
+		}
+	}
+
+	public function getParserOutputVars() {
+		return [
+			// 1st test case
+			[
```

Comment:
you can use named array item (specific string as key) you can make it easier to debug (the name appears in the test failure information